internal/toproto5/importresourcestate.go

```diff
@@ -33,5 +34,11 @@ func ImportResourceStateResponse(ctx context.Context, fw *fwserver.ImportResourc
 		proto5.ImportedResources = append(proto5.ImportedResources, proto5ImportedResource)
 	}
 
+	if fw.Deferred != nil {
+		proto5.Deferred = &tfprotov5.Deferred{
+			Reason: tfprotov5.DeferredReason(fw.Deferred.Reason),
+		}
+	}
```

Comment:
nit: We can probably break this into it's own file/function for `Deferred` in `toproto{5/6}`. Makes it easier to add more fields later 👍🏻 

Similar to `terraform-plugin-go`: https://github.com/hashicorp/terraform-plugin-go/blob/main/tfprotov5/internal/toproto/deferred.go

Reply:
It still seems like this feedback might be valid, even if that means having two separate `DataSourceDeferred()` and `ResourceDeferred()`, e.g.

```go
func ResourceDeferred(fw *resource.Deferred) *tfprotov5.Deferred {
  if fw == nil {
    return nil
  }

  return &tfprotov5.Deferred{
    Reason: tfprotov5.DeferredReason(fw.Deferred.Reason)
  }
}
```

As Austin mentions, this continues the pattern of having shared logic for the terraform-plugin-go type conversions.